### PR TITLE
BIP 80 & 81: Hierarchies for Voting Pool Deterministic Multisig Wallets

### PR DESCRIPTION
**Pull request by justusranvier**
Two informational BIPs are submitted for voting pool HD wallets.

This wallet format follows the BIP43 recommendation to use reserved BIP numbers to avoid namespace collisions.

The purpose of the wallet formats is to efficiently implement multisig, FIFO cold storage for bulk bitcoins and for colored bitcoins.

### CODE REVIEW

File: bip-0081.mediawiki [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+Public/private keypairs are numbered from index 0 in sequentially increasing manner. This number is used as child index in BIP32 derivation.
+
+Public keys obtained at this level of the heirarchy are used to construct multisig deposit scripts, using a schema that is shared between the members as an out-of-band contract.
+
+Public derivation is used at this level.
+
+==Compatible wallets==
+
+* [[https://github.com/conformal/btcd|btcd]] is the reference Bitcoin wallet for voting pools.
```
Comment (UdjinM6) [inline bip-0081.mediawiki]: Quote from https://github.com/btcsuite/btcd
> One key difference between btcd and Bitcoin Core is that btcd does NOT include wallet functionality and this was a very intentional design decision.

Should mention https://github.com/btcsuite/btcwallet here instead imo

File: bip-0081.mediawiki [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+==Reference==
+
+* [[bip-0032.mediawiki|BIP32 - Hierarchical Deterministic Wallets]]
+* [[bip-0043.mediawiki|BIP43 - Purpose Field for Deterministic Wallets]]
+* [[bip-0044.mediawiki|BIP44 - Multi-Account Hierarchy for Deterministic Wallets]]
+* [[bip-0080.mediawiki|BIP44 - Hierarchy for Non-Colored Voting Pool Deterministic Multisig Wallets]]
```
Comment (UdjinM6) [inline bip-0081.mediawiki]: This reference is wrong (should say "BIP**80** - ...") and it's never being used in the text above actually so I guess it should be removed.

File: bip-0081.mediawiki (hunk 1 above)
Comment (justusranvier) [inline bip-0081.mediawiki]: Thanks. Fixed this in the most recent commit.

File: bip-0081.mediawiki [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+==Motivation==
+
+The hierarchy proposed in this paper allows the handling of multiple color definitions from a single seed.
+
+==Path levels==
+
+We define the following 8 levels in BIP32 path:
+
+<pre>
+m / purpose' / series' / (5 color definition levels) / address_index
```
Comment (UdjinM6) [inline bip-0081.mediawiki]: Why there is no coin type for colored coins?

**Comment by luke-jr** [comment]
@justusranvier Can you add the Copyright sections?

**Comment by justusranvier** [comment]
Since my contract with them is ended, I no longer have access to the Monetas repo from which pull request originated.

If they are not interested in maintaining this BIP, I'll resubmit it from another organization.

**Comment by luke-jr** [comment]
@justusranvier If you own copyright to these files (please be sure they aren't considered work-made-for-hire and therefore the company owns them!), just ACKing a specific license/PD here is sufficient and I can add it in the merge...

**Comment by justusranvier** [comment]
I'm not sure what their status is, but here's my best guess:

The are for sure a work for hire, but the original version of the documents was released under an AGPL-3 license:

https://github.com/Open-Transactions/rfc/tree/master/bips

The documents in this pull request are derived from that. Monetas probably has the right to release them under any additional license they wish - anyone except them is limited to AGPL-3 terms for derived works.

Since their work on implementing this in btcwallet is still ongoing: https://github.com/btcsuite/btcwallet/pull/345, then probably the best person to ask about the status of these bips is @al-maisan or @gsalgado.

**Comment by gsalgado** [comment]
I can add that. Should I also change your email address, @justusranvier, since your @monetas.net one probably no longer works?

**Comment by luke-jr** [comment]
I think I can merge this as a Draft, but since BIP 1 does not allow for AGPL, it can't proceed until either the license is changed, or a BIP allowing more licenses becomes active.

How exactly is AGPL intended to be applicable to something like a BIP, btw?

**Comment by gsalgado** [comment]
I don't know what was the rationale for using AGPL-3 for those BIPs, but I just spoke to @al-maisan and Monetas is ok with placing them on the public domain, so I've updated the copyright sections to reflect that

**Pull request merged**